apps/src/schoolInfo/SchoolInfoInterstitial.jsx

```diff
@@ -1,378 +1,173 @@
-    };
-  }
+  }, []);
 
-  isBlank(field) {
-    // return true when field is blank
-    return !!(!field || field.trim() === '');
-  }
-
-  validateSubmission = () => {
-    const {country, schoolType, schoolName, ncesSchoolId} = this.state;
-    let errors = {};
-    let isValid = true;
-
-    if (this.isBlank(country)) {
-      errors.country = true;
-      isValid = false;
-    } else if (country === 'United States') {
-      if (this.isBlank(schoolType)) {
-        errors.schoolType = true;
-        isValid = false;
-      } else {
-        /**
-         * NCES (National center for education statistics) only stores information
-         * for private, public and charter schools in the United States.
-         * Teachers with NCES school IDs are unique becuase of the useful information
-         * that is provided by NCES that the data team can join onto/use such as the
-         * the number of students in a school, % of URM students.
-         */
-        const ncesSchoolType = ['public', 'private', 'charter'];
-        if (ncesSchoolType.includes(schoolType)) {
-          if (this.isBlank(ncesSchoolId)) {
-            errors.ncesSchoolId = true;
-            isValid = false;
-          }
-
-          // ncesSchoolId is set to -1 when the checkbox for school not found is clicked
-          // For a US, NCES school type, No NCES school id, school name is required.
-          if (ncesSchoolId === '-1' && this.isBlank(schoolName)) {
-            errors.schoolName = true;
-            isValid = false;
-          }
-        }
-      }
-    }
-    return {
-      errors,
-      isValid,
-    };
-  };
-
-  handleSchoolInfoSubmit = () => {
-    const {errors, isValid} = this.validateSubmission();
-    this.setState({
-      errors,
-    });
-    if (!isValid) {
-      return;
-    }
+  const handleSchoolInfoSubmit = async () => {
     analyticsReporter.sendEvent(
       EVENTS.SCHOOL_INTERSTITIAL_SUBMIT,
       {
-        hasNcesId:
-          this.state.ncesSchoolId && this.state.ncesSchoolId !== '-1'
-            ? 'true'
-            : 'false',
-        attempt: this.state.showSchoolInfoUnknownError ? 2 : 1,
+        hasNcesId: schoolInfo.schoolId ? 'true' : 'false',
+        attempt: showSchoolInfoUnknownError ? 2 : 1,
       },
       PLATFORMS.BOTH
     );
 
-    const schoolData = this.buildSchoolData();
-    const {formUrl, authTokenName, authTokenValue} = this.props.scriptData;
-    $.post({
-      url: formUrl + '.json',
-      dataType: 'json',
-      data: {
-        _method: 'patch',
-        [authTokenName]: authTokenValue,
-        ...schoolData,
-      },
-    })
-      .done(() => {
-        analyticsReporter.sendEvent(
-          EVENTS.SCHOOL_INTERSTITIAL_SAVE_SUCCESS,
-          {
-            attempt: this.state.showSchoolInfoUnknownError ? 2 : 1,
-          },
-          PLATFORMS.BOTH
-        );
+    try {
+      if (schoolInfo.schoolId === NO_SCHOOL_SETTING) {
+        onClose();
+        return;
+      }
```

Comment:
unsure about this. I didn't see a good way to store this for the user

Reply:
@hannahbergam confirmed all that needs to be sent is an analytics reporter event, which is done in `useSchoolInfo`